bundles/org.openhab.binding.bmwconnecteddrive/src/main/java/org/openhab/binding/bmwconnecteddrive/internal/discovery/VehicleDiscovery.java

```diff
@@ -0,0 +1,227 @@
+                                    + vehicle.dealer.postalCode + " " + vehicle.dealer.city);
+                            properties.put("dealerPhone", vehicle.dealer.phone);
+                        }
+
+                        // Services & Support
+                        properties.put("servicesActivated", getObject(vehicle, Constants.ACTIVATED));
+                        String servicesSupported = getObject(vehicle, Constants.SUPPORTED);
+                        String servicesNotSupported = getObject(vehicle, Constants.NOT_SUPPORTED);
+                        if (vehicle.statisticsAvailable) {
+                            servicesSupported += Constants.STATISTICS;
+                        } else {
+                            servicesNotSupported += Constants.STATISTICS;
+                        }
+                        properties.put(Constants.SERVICES_SUPPORTED, servicesSupported);
+                        properties.put("servicesNotSupported", servicesNotSupported);
+                        properties.put("supportBreakdownNumber", vehicle.breakdownNumber);
+
+                        // Vehicle Properties
+                        if (vehicle.supportedChargingModes != null) {
+                            StringBuilder chargingModes = new StringBuilder();
+                            vehicle.supportedChargingModes.forEach(e -> {
+                                chargingModes.append(e).append(Constants.SPACE);
+                            });
+                            properties.put("vehicleChargeModes", chargingModes.toString());
+                        }
+                        if (vehicle.hasAlarmSystem) {
+                            properties.put("vehicleAlarmSystem", "Available");
+                        } else {
+                            properties.put("vehicleAlarmSystem", "Not Available");
+                        }
+                        properties.put("vehicleBrand", vehicle.brand);
+                        properties.put("vehicleBodytype", vehicle.bodytype);
+                        properties.put("vehicleColor", vehicle.color);
+                        properties.put("vehicleConstructionYear", Short.toString(vehicle.yearOfConstruction));
+                        properties.put("vehicleDriveTrain", vehicle.driveTrain);
+                        properties.put("vehicleModel", vehicle.model);
+                        if (vehicle.chargingControl != null) {
+                            properties.put("vehicleChargeControl", Converter.toTitleCase(vehicle.model));
+                        }
+
+                        // Check now if a thing with the same VIN exists
+                        final AtomicBoolean foundVehicle = new AtomicBoolean(false);
+                        List<VehicleHandler> l = ConnectedDriveHandlerFactory.getHandlerRegistry();
+                        l.forEach(handler -> {
+                            Thing vehicleThing = handler.getThing();
```

Comment:
You can get all of the handlers by iterating through the handlers of the children of the bridge.
```suggestion
                        bridgeHandler.get().getThing().getThings().forEach(vehicleThing ->
```

Reply:
fixed. Replaced by 'bridgeHandler.ifPresent(...' at begin of method. All other code in method is pointless without valid bridgeHandler